### PR TITLE
Implemented hashUtil + password encryption on all layer where needed

### PR DESCRIPTION
**Pull request by andriylagun**

### CODE REVIEW

File: src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java [hunk 1]
```diff
@@ -15,7 +16,8 @@
     @Override
     public User login(String login, String password) throws AuthException {
         Optional<User> user = userService.findByLogin(login);
-        if (user.isEmpty() || !password.equals(user.get().getPassword())) {
+        if (!HashUtil.hashPassword(password, user.get().getSalt())
```
Comment (andromash) [inline src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java]: Why did you remove isEmpty check? is it 100% we will have users in Optional? And maybe it would be better to extract the logic of checking hashes passwords to separate method?

File: src/main/webapp/resources/init_db.sql [hunk 2]
```diff
@@ -15,6 +15,7 @@ CREATE TABLE `users`
     `login`    VARCHAR(225)   NOT NULL,
     `password` VARCHAR(225)   NOT NULL,
     `deleted`  TINYINT        NOT NULL DEFAULT true,
+    `salt`     VARBINARY(500) NOT NULL,
```
Comment (andromash) [inline src/main/webapp/resources/init_db.sql]: Why the size is 500? We definitely know that we will have an array of 16 elements. Or I misunderstood that?

File: src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java (hunk 1 above)
Comment (andriylagun) [reply src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java]: Thanks. missed that.

**Review by andromash — APPROVED** [review]
Well done!

**Review by andromash — APPROVED** [review]

File: src/main/webapp/resources/init_db.sql (hunk 2 above)
Comment (andriylagun) [reply src/main/webapp/resources/init_db.sql]: we have array of 16 bytes. so it will be more than 16 symbols

Comment (dmytro-hryhoruk) [reply src/main/webapp/resources/init_db.sql]: this number is the size in bytes so you can write like this VARBINARY(16)

Comment (dmytro-hryhoruk) [inline src/main/webapp/resources/init_db.sql]:
```suggestion
    `salt`     VARBINARY(16) NOT NULL,
```
should work like this

**Review by dmytro-hryhoruk — APPROVED** [review]
